ex02.exs

```diff
@@ -1,6 +1,24 @@
 
 defmodule Ex02 do
 
+  @name __MODULE__
+
+  def new_counter(value \\ 0) do
+    { :ok, _ } = Agent.start_link(fn -> value end)
+  end
+
+  def next_value({ :ok, counter }) do
+    Agent.get_and_update(counter, fn (count) -> { count, count + 1 } end)
+  end
+
+  def new_global_counter(value \\ 0) do
+    Agent.start_link(fn -> value end, name: @name)
+  end
+
+  def global_next_value do
```

Comment:
## +2

for being the first person so far to eliminate duplication by calling next_value from the global variant.